### PR TITLE
feat: added labels and tags fetching for BigQuery

### PR DESCRIPTION
**Pull request by manuelzander**
- Using `datacatalog_v1` Python client to fetch tags for a given BQ table
- Added the tags data to `TableMetadata` and `ColumnMetadata`  where appropriate
- Also added labels to `TableMetadata`

### CODE REVIEW

File: pipelines/whale/extractor/bigquery_metadata_extractor.py [hunk 1]
```diff
@@ -79,9 +100,15 @@ def _retrieve_tables(self, dataset) -> Any:
...
                                 total_cols = self._iterate_over_cols(
-                                    "", column, cols, total_cols + 1
+                                    tags_dict, "", column, cols, total_cols + 1
                                 )
 
+                table_tag = None
+                if "tags" in tags_dict:
+                    for tag in tags_dict["tags"]:
+                        if not "column" in tag:
+                            table_tag = tag
```
Comment (rsyi) [inline pipelines/whale/extractor/bigquery_metadata_extractor.py:110]: Huh. I didn't know you could do `not x in y`. I always write it `x not in y`. Learning something new every day :)

File: pipelines/tests/unit/extractor/test_bigquery_metadata_extractor.py [hunk 2]
```diff
@@ -430,10 +587,14 @@ def test_keypath_and_pagesize_can_be_set(self, mock_build):
             )
 
     @patch("whale.extractor.base_bigquery_extractor.build")
-    def test_table_part_of_table_date_range(self, mock_build):
-        mock_build.return_value = MockBigQueryClient(
+    @patch("whale.extractor.base_bigquery_extractor.datacatalog_v1")
+    def test_table_part_of_table_date_range(self, mock_datacatalogue, mock_bigquery):
```
Comment (rsyi) [inline pipelines/tests/unit/extractor/test_bigquery_metadata_extractor.py:591]: BRITISH ENGLISH!? UNACCEPTABLE!
Just kidding, please don't change it. :)

**Review by rsyi — COMMENTED** [review]
LGTM already, @manuelzander. Thanks for tackling this. :) Let me know when it's ready to come out of Draft, and I'll run my own local tests and merge.

File: pipelines/tests/unit/extractor/test_bigquery_metadata_extractor.py (hunk 2 above)
Comment (manuelzander) [reply pipelines/tests/unit/extractor/test_bigquery_metadata_extractor.py:591]: Oh haha my bad, I mixed it up 😂  [Google style...](https://github.com/googleapis/python-datacatalog/)

File: pipelines/whale/extractor/bigquery_metadata_extractor.py (hunk 1 above)
Comment (manuelzander) [reply pipelines/whale/extractor/bigquery_metadata_extractor.py:110]: I could change it to make it clearer for everybody :)

**Comment by manuelzander** [comment]
> LGTM already, @manuelzander. Thanks for tackling this. :) Let me know when it's ready to come out of Draft, and I'll run my own local tests and merge.

@rsyi Are you happy with labels being stored as `dict` e.g. `{"label_1": "this is a label", ...}`, and tags as `Tag` structure, see [official JSON representation](https://cloud.google.com/data-catalog/docs/reference/rest/v1/projects.locations.entryGroups.entries.tags#Tag)?
@biomunky FYI

**Comment by rsyi** [comment]
Yeah I was wondering about this too, but it seemed like bigquery tags are more like weird, nested json annotations than they are plain string tags, so we can't just throw away everything but the field names...

I think this is fine for now. Another option (though a bit more work) could be to create a separate `BigqueryTagExtractor` class that dumps this data either into a `BigqueryTableTag` or `BigqueryColumnTag` model, but feel free to punt on that if it feels like overkill. I'm not one for over-abstracting when it's unlikely the abstraction will ever serve a second purpose in the future. :)

**Comment by manuelzander** [comment]
@rsyi I've just added an additional test (no tags for a given BQ table).
I'm happy with this to be merged now. Having something like a `BigqueryTagExtractor` could be useful in case there would be Spanner tags as well - so maybe rather a `GoogleCloudTagExtractor` - BUT apparently Spanner does currently not support tags natively, so we don't really need this for now 😊

**Pull request merged**